### DISCUSSION TITLE
Implement `Related<T>` trait using `Relation` with `DeriveRelation` macro

### DISCUSSION THREAD

**Discussion by LasseRosenow** [SeaQL/sea-orm#1148] [Ideas]
Right now the relation between tables is defined a bit redundant, or maybe rather "overly complicated".
First we have an `Relation` enum, which specifies all the relations of our `entity`.
But then we still have to implement the `Related<T>` trait for each relation, referencing to our `Relation` enum.

This creates a lot of boilerplate, which is why I propose to automatically generate the `Related<T>` trait implementation using the `DeriveRelation` macro.

I am not sure if this is a language limitation and maybe my proposed solution is not possible. I am not advanced in my knowledge about Rust to answer this question, so I would like to ask, if somebody knows better than me:

### Code Example:
As you can see we need to explicitly write the Related<T> trait implementation, even though all the information for it would already be defined in the macro above:
```rust
#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {
    #[sea_orm(has_many = "super::fruit::Entity")]
    Fruit,
}

impl Related<super::fruit::Entity> for Entity {
    fn to() -> RelationDef {
        Relation::Fruit.def()
    }
}
```
What I would like is, that the second part could be automatically generated like this:
```rust
#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {
    #[sea_orm(has_many = "super::fruit::Entity")]
    Fruit,
}
```

**Comment by billy1624** [comment]
Hey @LasseRosenow, welcome!! This is actually part of the design.

Consider two table with multiple foreign key linkage. How would SeaORM implement `Related<T>` for them? This is a choice that has to be made by the user but not by the SeaORM.

Does this make sense?
